qiskit_qec/linear/symplectic.py

```diff
@@ -50,7 +50,7 @@ def all_commute(matrix: np.ndarray) -> bool:
         >>> all_commute(matrix)
         True
     """
-    test_mat = symplectic_product(matrix, matrix)
+    test_mat = np.asarray(symplectic_product(matrix, matrix))
```

Comment:
`all_commute` takes in a matrix and `symplectic_product(matri,matrix)` will in general return a numpy array of values It will only return a `int` when matrix is a single row (i.e. vector)

Reply:
Oh, then we should keep this but we need to update the outdated `symplectic_product` function `->`  because it's `-> int` right now. 

<img width="516" alt="Screen Shot 2022-07-19 at 10 47 30 PM" src="https://user-images.githubusercontent.com/74260313/179885212-e38f10e1-ec47-4750-a761-5b9ecf03fcfd.png">

Reply:
Good catch. Yes. Let's change this to `Union[int, np.ndarray]`